pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/generators/TestGenerator.java

```diff
@@ -292,6 +292,14 @@ private static String generateTestMethod(final String modelClassName, final Stri
 					// check fast equals
 					dataBuffer2.resetPosition();
 					$modelClassName.PROTOBUF.fastEquals(modelObj, dataBuffer2);
+
+					// Test toBytes()
+					Bytes bytes = $modelClassName.PROTOBUF.toBytes(modelObj);
+					final var dataBuffer3 = getThreadLocalDataBuffer();
+					dataBuffer3.resetPosition();
+					final int protoBufByteCount1 = (int)dataBuffer3.remaining();
+					bytes.toReadableSequentialData().readBytes(dataBuffer3);
+					$modelClassName.PROTOBUF.fastEquals(modelObj, dataBuffer3);
```

Comment:
Think this could be much simpler like:
```
/ / Test toBytes()
assertEquals(dataBuffer,$modelClassName.PROTOBUF.toBytes(modelObj));
```

Reply:
I wasn't sure if there will be "solution" reusing the object (there are multiple objects used, so I did the same). Changed.

Reply:
So, it looks like reusing of buffer is not a good idea. All the tests are failing. The issue is that the buffer is already polluted by another test and has other data. Starting with a fresh buffer (as dataBuffer2 is done) makes everything work.

Reply:
does it not need `assertTrue($modelClassName....` as well

Reply:
Ops... Too much Undo in the editor. Good catch.

Reply:
Actually, chnge it to actually get the byte[] array and compare that. This way we don't need to call fastEquals() and not find errors if that function has bugs.